GWFish/modules/inspiral_corrections.py

```diff
@@ -0,0 +1,912 @@
+#Python libraries
+
+import os
+import logging
+import requests 
+from io import StringIO
+import matplotlib.pyplot as plt
+import numpy as np
+import sympy as sp
+from scipy.interpolate import interp1d
+import scipy.optimize as optimize
+
+#LAL libraries
+
+try:
+    import lalsimulation as lalsim
+    import lal
+    from lal import CreateREAL8Vector
+except ModuleNotFoundError as err:
+    uselal = err
+    logging.warning('LAL package is not installed.'+\
+                    'Only GWFish waveforms available.')
+
+#GWFish libraries
+
+import GWFish as gw 
+import GWFish.modules.constants as cst
+import GWFish.modules.auxiliary as aux
+import GWFish.modules.fft as fft
+
+################################################################################
+########################### Some FUNCTION DEFINITIONS ##########################
+################################################################################
+
+def convert_args_list_to_float(*args_list):
+    """
+    Converts inputs to floats, returns a list in the same order as the 
+    input.
+    """
+    # copied from https://git.ligo.org/lscsoft/bilby/, March 21, 2022
+    try:
+        args_list = [float(arg) for arg in args_list]
+    except ValueError:
+        raise ValueError("Unable to convert inputs to floats")
+    return args_list
+
+def t_of_f_PN(parameters, frequencyvector):
+    """
+    t(f) is required to calculate slowly varying antenna pattern as a 
+    function of instantaneous frequency. This FD approach follows 
+    Marsat/Baker arXiv:1806.10734v1; equation (22) neglecting the phase 
+    term, which does not matter for SNR calculations.
+    """
+
+    local_params = parameters.copy()
+    aux.check_and_convert_to_mass_1_mass_2(local_params)
+
+    M1 = local_params['mass_1'] * cst.Msol
+    M2 = local_params['mass_2'] * cst.Msol
+
+    M = M1 + M2 #Total mass
+    mu = M1 * M2 / M #Reduced mass
+    Mc = cst.G * mu ** 0.6 * M ** 0.4 / cst.c ** 3 # Chirp mass in seconds
+
+    #t(f)=(1/2pi)*phi_prime(f) at lowest order in stationary phase approximation
+    t_of_f = -5./(256.*np.pi**(8/3))/Mc**(5/3)/frequencyvector**(8/3)
+
+    return t_of_f+parameters['geocent_time']
+
+################################################################################
+################### WAVEFORMS PARAMETERS AND VARIABLES #########################
+################################################################################
+
+class Waveform:

```

Comment:
Is this a redefinition of the Waveform class from the `waveforms.py` module? If so, that's dangerous... If you need to modify its functionality you can use inheritance (or we can discuss and change the original one)

Reply:
No, there is no redefinition of the Waveform class. The only modification is regarded to the addition of the ppE parameters and f_cut in self.gw_params. Maybe I could modify this directly in waveforms.py